added: hsl to rgb conversion

AndreyDmitriev39r opened the pull request.

jdwilkin4 commented:
@AndreyDmitriev39r 
For draft PR's, you don't need to use the Do not merge label since GitHub automatically disable the merge button anyway
<img width="464" alt="Screen Shot 2022-08-10 at 1 50 04 PM" src="https://user-images.githubusercontent.com/67210629/184017443-a431fc24-9c19-4120-beed-4759c207bced.png">

jdwilkin4 commented:
> So I have conversion from HSL to RGB now.

This seems out of scope for this ticket though. 
Even though your function works, we want to make sure to focus on the main task first. 
Remember the primary goal for this ticket is for users to input two hsl and hsla values and get back a contrast ratio.

We should focus on getting basic inputs in place
two rgb values
two hsl values
two hex values

In future tickets, we can worry about hsl to rgb
hex to rgb and etc. 

Hope that helps! 👍

AndreyDmitriev39r commented:
I converted HSL to RGB, because
We already have function to calculate luminance. It uses as input RGB values.
So before calculating ratio for HSL I converted it to RGB, then I will use existing codebase to calculate ratio.

jdwilkin4 commented:
Ok, I see what you are doing now.

>What's next? Should I write new function like twoHSLRatio()?
>Or should I modify existing twoHexesRatio() to handle stuff for HSL also?
>Or maybe first I need to deal with HSLA to RGB conversion?

Why don't you do the twoHSLRatio next and test displaying the contrast ratio output.
Once that is working, then you can add the HSLA to RGB conversion.

Then we can go from there and refactor as needed 👍

AndreyDmitriev39r commented:
Have functionality for displaying result for two HSL colors.
However, look how i broke this one:

https://contrast-ratio.com/#hsl%28223%2C%2015%25%2C%2027%25%29-on-hsl%2810101010101101%2C%20415%25%2C%2029%25%29

Should I be bothered wuth checking numbers from user for correctness at this point?
for example, Hue(1 number) doesn't make sense if its more than 360.

I see the merge conflict, but I don't care for now - still need to deal with HSLA part of the task.

jdwilkin4 commented:
@AndreyDmitriev39r 

> Should I be bothered with checking numbers from user for correctness at this point?

Let's first build it out assuming the user inputs the correct format and information.
Once we get all of the basics working, then we can focus on error handling and what to do if a use inputs the wrong information.

> We have bunch of repetitions in the code at this point:
>look at twoHexesRatio and twoHSLratio funcs.
>Should I be bothered with this right nowin this ticket , or we can refactor it all at once, when we have functionality for more cases?

Let's focus on getting the basic functionality working for the rgb,hsl and hex codes.
Then we can have one ticket that focuses on refactoring the code for those three formats

AndreyDmitriev39r commented:
Couple of moments to keep in mind:

1. Current *hsla contrast ratio functionality* highly depends on existing *rgba contrast ratio functionality*. It appears more research needed to make all semi-transparent stuff work 100% right.
2. Right now _swatch dynamic for hsl and hsla_ will work if input will be: `hsl(225, 30%, 30%)` or `hsla(225, 30%, 30%, 0.6)`. Accent on `%` symbols. Without them (for example `hsl(225, 30, 30)`) - contrast ratio will be calculated, BUT swatch colors will not be displayed correctly.

sparkO7 reviewed (approved):
LGTM!

Okibe00 commented:
Looks good, I guess we will be refactoring later on.
